src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,126 @@
+import games.strategy.triplea.delegate.dataObjects.BattleRecord;
+import games.strategy.triplea.delegate.remote.IBattleDelegate;
+import games.strategy.triplea.formatter.MyFormatter;
+import games.strategy.triplea.oddsCalculator.ta.BattleResults;
+import games.strategy.triplea.player.ITripleAPlayer;
+import games.strategy.util.CompositeMatch;
+import games.strategy.util.CompositeMatchAnd;
+import games.strategy.util.CompositeMatchOr;
+import games.strategy.util.IntegerMap;
+import games.strategy.util.Match;
+import games.strategy.util.Tuple;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // we may start multiple times due to loading after saving
+    // only initialize once
+
+    
+    IBattle battle;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(true) ) {   // loop throught bombing/air raids
+      battle = m_battleTracker.getPendingBattle(t, true, null);
+      if( battle == null ) {
+        try {
+          throw new Exception("Air/Bombing Raid gone missing in BattleDelegate");
+        } catch( Exception e ) {  // Will crash below
+        }
+      }
+      battle.fight(m_bridge);
+      battle = m_battleTracker.getPendingBattle(t, true, BattleType.BOMBING_RAID );  // check to see if there's still a bombing raid for the territory - i.e. previous battle was an air raid
+      if( battle != null ) {
+        battle.fight(m_bridge);
+      }
+    }
+
+    int otherBattleCount = 0;
+    int amphibCount = 0;
+    IBattle lastAmphib = null;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Loop through normal combats i.e. not bombing or air raid
+      battle = m_battleTracker.getPendingBattle(t, false, BattleType.NORMAL);
+      // we only care about battles where we must fight
+      // this check is really to avoid implementing getAttackingFrom() in other battle subclasses
+      if (!(battle instanceof MustFightBattle) ) {
+        continue;
+      }
+      if( !t.isWater() ) {
+        final Map<Territory, Collection<Unit>> attackingFromMap = ((MustFightBattle) battle).getAttackingFromMap();
+        for( final Territory neighbor : ((MustFightBattle) battle).getAttackingFrom() ) {
+          if (!neighbor.isWater() || Match.allMatch(attackingFromMap.get(neighbor), Matches.UnitIsAir) ) {
+            continue;
+          }
+          amphibCount++;
+          lastAmphib = battle;
+          break;
+        }
+      }
+      if( lastAmphib != battle ) {
+        otherBattleCount++;
+      }
+    }
+
+    if( amphibCount > 1 ) {
+      return;
+    }
+
+    // Fight amphibious assault if there is one. Fight naval battles in random order if prerequisites first.
+    if( amphibCount > 0 ) {
+      if( m_currentBattle != null && m_currentBattle != lastAmphib ) {      // Not completely sure if this is needed but was in other code and does no harm
```

Comment:
I'm happy you documented the uncertainty, but we are leaving an unresolved problem for yet another day, and worse yet, we're scattering it to more parts of the code.

Please find ways to extract common code, and re-use it. 

IE:
```
private static class AmphibBattleCounter {
  public int count(m_battleTracker) {
      :
      :
  }
}
```